src/premint/EIP712UpgradeableWithChainId.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
```

Comment:
This is a modified version of OpenZeppelin's EIP712Upgradeable that allows a chain id to be explicitly passed when generating the hash